### PR TITLE
Added paragraph about pod restart control

### PR DESCRIPTION
**Pull request by ANeumann82**
Signed-off-by: Andreas Neumann <aneumann@mesosphere.com>

### CODE REVIEW

File: content/docs/developing-operators/tasks.md [hunk 1]
```diff
@@ -32,6 +32,12 @@ KUDO will apply all the listed resources and wait for all of them to become heal
...
 
+The apply-task enhances deployed resources with KUDO specific labels and annotations. For resources that create or deploy pods (StatefulSets, Deployments, Daemonsets, etc. ) KUDO adds a special annotation to the pod template spec that is the hash from all resources used by the pod template spec (ConfigMaps and Secrets). The effect of this hash is that an update to an operator parameter that changes a ConfigMap triggers a restart of the pods that uses that ConfigMap.
+
+**Note:** This pod restart control will happen when the parent resource (StatefulSet, Deployment) is applied the next time. This can happen in the same task as the dependency, a task that is executed later in the same plan, or in a different plan executed at a later time. The pod restarts will not happen until the parent resource is applied again.
```
Comment (nfnt) [inline content/docs/developing-operators/tasks.md]: After reading this paragraph a few times, I'm still not sure what is stated here.

**Review by nfnt — COMMENTED** [review]
This is great, but I find the second paragraph hard to comprehend. Is it saying that pods will only be restarted if resources affecting this pod (e.g. a ConfigMap) is updated?

File: content/docs/developing-operators/tasks.md (hunk 1 above)
Comment (ANeumann82) [reply content/docs/developing-operators/tasks.md]: Rephrased that, hope it's a little bit clearer?

**Review by zen-dog — APPROVED** [review]
SGTM 🚢

**Review by nfnt — APPROVED** [review]
Much clearer, thank you! 🚢

File: content/docs/developing-operators/tasks.md (hunk 1 above)
Comment (nfnt) [reply content/docs/developing-operators/tasks.md]: Yes, got it now, thank you!

**Pull request merged**